### PR TITLE
impl(pubsub): add exactly once delivery enabled attribute to subscribe span

### PR DESCRIPTION
**Pull request by alevenberg**
[#13720](https://github.com/googleapis/google-cloud-cpp/issues/13720)

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13889)

### CODE REVIEW

File: google/cloud/pubsub/internal/tracing_batch_callback_test.cc [hunk 1]
```diff
@@ -165,6 +164,41 @@ TEST(TracingBatchCallback, SubscribeAttributesForOrderingKey) {
...
+TEST(TracingBatchCallback, SubscribeAttributesForExactlyOnce) {
+  auto span_catcher = InstallSpanCatcher();
+  auto mock = std::make_shared<pubsub_testing::MockBatchCallback>();
+  EXPECT_CALL(*mock, callback).Times(1);
+  auto batch_callback = MakeTestBatchCallback(std::move(mock));
+  std::shared_ptr<opentelemetry::context::propagation::TextMapPropagator>
+      propagator = std::make_shared<
+          opentelemetry::trace::propagation::HttpTraceContext>();
+  BatchCallback::StreamingPullResponse response;
+  google::pubsub::v1::StreamingPullResponse r;
```
Comment (devbww) [inline google/cloud/pubsub/internal/tracing_batch_callback_test.cc:176]: Nit:  I would delay instantiating these objects until just before you need them.

File: google/cloud/pubsub/internal/tracing_batch_callback_test.cc [hunk 2]
```diff
@@ -165,6 +164,41 @@ TEST(TracingBatchCallback, SubscribeAttributesForOrderingKey) {
...
+  auto span = MakeTestSpan();
+  auto message = pubsub::MessageBuilder().Build();
+  InjectTraceContext(message, *propagator);
+  span->End();
+  auto proto_message = ToProto(message);
+  proto_message.set_message_id("id-0");
+
+  auto* m = r.add_received_messages();
+  r.mutable_subscription_properties()->set_exactly_once_delivery_enabled(true);
+  *m->mutable_message() = proto_message;
```
Comment (devbww) [inline google/cloud/pubsub/internal/tracing_batch_callback_test.cc:186]: Nit:  Similarly, I would localize `m`'s initialization with its use.

**Review by devbww — APPROVED** [review]

File: google/cloud/pubsub/internal/tracing_batch_callback_test.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_batch_callback_test.cc]: Done

File: google/cloud/pubsub/internal/tracing_batch_callback_test.cc (hunk 2 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_batch_callback_test.cc]: Done

**Pull request merged**